### ISSUE TITLE
Class literals for existential value types do not match the JVM

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#2801]
The test `RegressionTest.should_support_class_literals_for_existential_value_types_issue_218` does not pass on the JVM. The code of the test is:
```scala
  @Test def should_support_class_literals_for_existential_value_types_issue_218(): Unit = {
    assertEquals("org.scalajs.testsuite.compiler.RegressionTest$Bug218Foo",
        scala.reflect.classTag[Bug218Foo[_]].toString)
  }
```
and the error is:
```
[error] Test org.scalajs.testsuite.compiler.RegressionTest.should_support_class_literals_for_existential_value_types_issue_218 failed: org.junit.ComparisonFailure: expected:<[org.scalajs.testsuite.compiler.RegressionTest$Bug218Foo]> but was:<[Object]>, took 0.001 sec
[error]     at org.scalajs.testsuite.compiler.RegressionTest.should_support_class_literals_for_existential_value_types_issue_218(RegressionTest.scala:148)
[error]     ...
```
This was initially implemented in #245 to fix the issue #218. Somehow we became smarter than Scala/JVM, since we can actually give a precise class literal for the value class, whereas Scala/JVM gives `Object`. We should be bug-compatible, though.

**Comment by hrhino** [comment]
@sjrd the PR for this just got merged (in case you didn't notice); should this get closed? Thanks!

**Comment by sjrd** [comment]
Will make sure whether we need to do anything (like adapting a test) when I'm back from vacation.

**Comment by sjrd** [comment]
Fixed in e259939196706f9a7d69e909db15e504b25f379d.
